File: handlers/start.py
from aiogram import types, Router, Dispatcher
from database.database import add_user, check_user
from utils.random_16 import random_string
from aiogram.filters import Command

router = Router()

@router.message(Command(commands=['start']))
async def send_welcome(message: types.Message):
    user_id = message.from_user.id
    full_name = message.from_user.full_name
    if not check_user(user_id):
        add_user(user_id, random_string(), 2)
        await message.answer(
            f'Hello, *{full_name}*!\nNice to meet you.\nYour profile has been *created* and we have replenished your balance with 2 lowi coins!\nTap /commands to start or /help to _... help message?_!',
            parse_mode="Markdown"
        )
    else:
        await message.answer(
            f'Welcome back, *{full_name}*!\nYour profile already exists. Tap /commands to start or /help to _... help message?_!',
            parse_mode="Markdown"
        )

def register_handlers(dp: Dispatcher):
    dp.include_router(router)

File: src/handlers/start.rs
use crate::db;
use crate::random_16::random_string;
use frankenstein::{Api, Message, SendMessageParams, TelegramApi};

pub async fn handle_start(api: &Api, message: Message) {
    // Extract user information from the message
    let user_id = message.from.as_ref().map(|u| u.id).unwrap_or(0);
    let full_name = message
        .from
        .as_ref()
        .map(|u| u.first_name.clone())
        .unwrap_or_else(|| "User".to_string());

    let exists = db::check_user(user_id as i64).unwrap_or(false);
    let response: String;
    if !exists {
        let new_idpay = random_string(16);
        if let Err(e) = db::add_user(user_id as i64, &new_idpay, 2) {
            response = format!("Error creating user: {}", e);
        } else {
            response = format!(
                "Hello, *{}*!\nNice to meet you.\nYour profile has been *created* and we have replenished your balance with 2 lowi coins!\nTap /commands to start or /help to _... help message?_!",
                full_name
            );
        }
    } else {
        response = format!(
            "Welcome back, *{}*!\nYour profile already exists. Tap /commands to start or /help to _... help message?_!",
            full_name
        );
    }

    let params = SendMessageParams {
        chat_id: frankenstein::ChatId::Integer(message.chat.id),
        text: response,
        parse_mode: Some("Markdown".to_string()),
        entities: None,
        disable_web_page_preview: None,
        disable_notification: None,
        reply_to_message_id: None,
        allow_sending_without_reply: None,
        reply_markup: None,
    };
    let _ = api.send_message(&params);
}
